contracts/Hub.sol

```diff
@@ -0,0 +1,123 @@
+ * @author The Everest team.
+ */
+
+contract Hub is AuthorizedAddresses {
+    // Contract addresses
+    ICollateralFunds public fundContract;
+    IStorage public storageContract;
+    IUniswapV3Factory public factory;
+
+    mapping(string => address) public tickersymbolToSynthAssetContractAddress;
+    mapping(string => address) public tickerSymbolToTradingPool;
+
+    address public USDCKovan = 0xe22da380ee6B445bb8273C81944ADEB6E8450422;
+    address public uniswapV3Factory = 0x1F98431c8aD98523631AE4a59f267346ea31F984;
+
+    /**
+     * @notice gets the synthAsset contract Address from its tickerSymbol
+     * @param tickerSymbol tickerSymbol to identify the synthAsset
+     * @return address address of synthAsset
+     */
+    function getSynthAssetContractAddress (string memory tickerSymbol) public view returns (address) {
+        return tickersymbolToSynthAssetContractAddress[tickerSymbol];
+    }
+
+    /**
+     * @notice mints synthAssets to a specific address
+     * @param receiver address to which token gets minted
+     * @param amount amount of token that gets minted
+     * @param tickerSymbol identifier of which token gets minted
+     */
+    function mintSynthAsset(address receiver, uint256 amount, string memory tickerSymbol) public onlyAuthorizedAddresses{
+        ISyntheticAsset(tickersymbolToSynthAssetContractAddress[tickerSymbol]).mint(receiver, amount); 
+    }
+
+    /**
+     * @notice sets the contract address of on sythetic asset (e.g synthTSLA)
+     * @param _synthAssetAddress contract address
+     */
+    function setSynthAssetContractAddress(address _synthAssetAddress , string memory tickerSymbol) public onlyOwner{
+        tickersymbolToSynthAssetContractAddress[tickerSymbol] = _synthAssetAddress ;
+    }
+
+    /**
+     * @notice sets the Fund contract address
+     * @param _fundAddress address of Fund contract
+     */
+    function setCollateralFundsContract(address _fundAddress) public onlyOwner{
+        fundContract = ICollateralFunds(_fundAddress);
+    }
+
+    /**
+     * @notice sets the Storage contract address
+     * @param _storageAddress address of Storage contract
+     */
+    function setStorageContract(address _storageAddress) public onlyOwner{
+        storageContract = IStorage(_storageAddress);
+    }
+
+    /**
+     * @notice checks the collateral ratio of an address.
+     * @dev for all assets combined. No individual / isolated positions for now.
+     * @dev price checking of colllateral too -> volatile collateral / depegging of stable collateral
+     * @param addr address of user whom collateral ratio is to be checked.
+     * @param collateralTickerSymbol identifier of token used for collateral#
+     * @return collateral ratio
+     */
+    function getCollateralRatioByAddress(address addr, string memory collateralTickerSymbol) public view returns (uint256) {
+        //Check amount funded
+        uint256 amountFunded = fundContract.getCollateralByAddress(msg.sender);
```

Comment:
`msg.sender` should be replaced with `addr` no? we want to know the collateral of a user not of the hub contract